src/Microsoft.Azure.ServiceBus/MessageReceivePump.cs

```diff
@@ -32,15 +32,9 @@ sealed class MessageReceivePump
             this.maxConcurrentCallsSemaphoreSlim = new SemaphoreSlim(this.registerHandlerOptions.MaxConcurrentCalls);
         }
 
-        public async Task StartPumpAsync()
+        public void StartPump()
         {
-            var initialMessage = await this.messageReceiver.ReceiveAsync(Constants.MessageReceiverStartPumpInitialReceiveTimeout).ConfigureAwait(false);
-            if (initialMessage != null)
-            {
-                MessagingEventSource.Log.MessageReceiverPumpInitialMessageReceived(this.messageReceiver.ClientId, initialMessage);
-            }
-
-            TaskExtensionHelper.Schedule(() => this.MessagePumpTask(initialMessage));
+            TaskExtensionHelper.Schedule(() => this.MessagePumpTaskAsync());
```

Comment:
I haven't spent time on 186, but since we are not changing that behavior here,  I would prefer to handle that as a separate issue itself.